code/game/objects/machinery/robotic_cradle.dm

```diff
@@ -0,0 +1,311 @@
+				reason = "Reason for discharge: Unauthorized manual release. Alerting security advised."
+			if(CRADLE_NOTICE_IDIOT_EJECT)
+				playsound(src.loc, 'sound/machines/warning-buzzer.ogg', 50, FALSE)
+				reason = "Reason for discharge: Unauthorized manual release during repair. Alerting security advised."
+			if(CRADLE_NOTICE_FORCE_EJECT)
+				playsound(src.loc, 'sound/machines/warning-buzzer.ogg', 50, FALSE)
+				reason = "Reason for discharge: Destruction of linked CRADLE Engineering System. Alerting security advised."
+		radio.talk_into(src, "<b>Patient: [occupant] has been released from [src] at: [get_area(src)]. [reason]</b>", RADIO_CHANNEL_MEDICAL)
+	occupant = null
+	update_icon()
+	stop_processing()
+
+/obj/machinery/robotic_cradle/proc/move_inside_wrapper(mob/living/dropped, mob/dragger)
+	if(dragger.incapacitated() || !ishuman(dragger) || dragger.get_species() == "Human")
+		return
+
+	if(occupant)
+		to_chat(dragger, span_notice("[src] is already occupied!"))
+		return
+
+	if(machine_stat & (NOPOWER|BROKEN))
+		to_chat(dragger, span_notice("[src] is non-functional!"))
+		return
+
+	if(dragger.skills.getRating(SKILL_ENGINEER) < SKILL_ENGINEER_ENGI && !event)
+		dropped.visible_message(span_notice("[dropped] fumbles around figuring out how to get into \the [src]."),
+		span_notice("You fumble around figuring out how to get into \the [src]."))
+		var/fumbling_time = max(0 , SKILL_TASK_TOUGH - ( SKILL_TASK_EASY * dragger.skills.getRating(SKILL_ENGINEER) ))// 8 secs non-trained, 5 amateur
+		if(!do_after(dropped, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+			return
+
+	dropped.visible_message(span_notice("[dropped] starts climbing into \the [src]."),
+	span_notice("You start climbing into \the [src]."))
+	if(do_after(dropped, 1 SECONDS, FALSE, src, BUSY_ICON_GENERIC))
+		if(occupant)
+			to_chat(dragger, span_notice("[src] is already occupied!"))
+			return
+		dropped.stop_pulling()
+		dropped.forceMove(src)
+		occupant = dropped
+		icon_state = "pod_0"
+		var/implants = list(/obj/item/implant/neurostim)
+		var/mob/living/carbon/human/H = occupant
+		var/doc_dat
+		med_scan(H, doc_dat, implants, TRUE)
+		start_processing()
+		for(var/obj/O in src)
+			qdel(O)
+
+	say("Automatic mode engaged, initialising procedure.")
+	addtimer(CALLBACK(src, PROC_REF(auto_start)), 20 SECONDS)
+
+///Callback to start auto mode on someone entering
+/obj/machinery/robotic_cradle/proc/auto_start()
+	if(repairing)
+		return
+	if(!occupant)
+		say("Occupant missing, procedures canceled.")
+	say("Beginning repair procedure.")
+	repair_op()
+
+/obj/machinery/robotic_cradle/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(!ishuman(user))
+		return // no
+
+	else if(istype(I, /obj/item/healthanalyzer) && occupant) //Allows us to use the analyzer on the occupant without taking him out.
+		var/obj/item/healthanalyzer/J = I
+		J.attack(occupant, user)
+		return
+
+	else if(!istype(I, /obj/item/grab))
```

Comment:
Elses unnecesarry here too. Also you're checking if the item is a grab twice for no reason.